docs/sources/distributed-traces.md

```diff
@@ -0,0 +1,79 @@
+weight: 5
+keywords:
+  - Beyla
+  - eBPF
+  - distributed traces
+aliases:
+  - /docs/grafana-cloud/monitor-applications/beyla/distributed-traces/
+---
+
+# Distributed traces with Beyla
+
+## Introduction
+
+Beyla currently supports distributed traces for Go applications, both HTTP/S and gRPC, with 
+some limitations and version restrictions. Other programming languages than Go are not supported
+yet.
+
+Go distributed tracing is implemented through propagation of the [W3C `traceparent`](https://www.w3.org/TR/trace-context/)
+header value. Beyla will read any incoming trace context header values, track the Go program execution flow and propagate
+the trace context by automatically adding the `traceparent` field in outgoing HTTP/gRPC requests. If your application
+already adds the `taceparent` field in outgoing requests, Beyla will use that value for tracing instead its own generated
+trace context. If Beyla cannot find an incoming `traceparent` context value, it will generate one according to the W3C
+specification.
+
+`Traceparent` context propagation is automatic and it doesn't require any action or configuration.
+
+## Limitations
+
+### Kernel integrity mode limitations
+
+In order to write the `traceparent` value in outgoing HTTP/gRPC request headers, Beyla needs to write in the process 
+user memory, which is currently done by using the [bpf_probe_write_user](https://www.man7.org/linux/man-pages/man7/bpf-helpers.7.html) eBPF helper. Since kernel 5.17, this helper is protected (and unavailable to BPF programs) if the Linux 
+Kernel is running in `integrity` lockdown mode. Kernel integrity mode is typically enabled by default if the Kernel has 
+[Secure Boot](https://wiki.debian.org/SecureBoot) enabled, but it can also be enabled manually. 
+
+Beyla will automatically check if it can use the `bpf_probe_write_user` helper, and enable context propagation only if it's
+allowed by the kernel configuration. You can verify your Linux Kernel lockdown mode by running the following command:
+
+```
+cat /sys/kernel/security/lockdown
+```
+
+If that file exists on your system and the mode is anything other than `[none]`, Beyla will not be able to perform
+context propagation and distributed tracing will be disabled.
+
+### Configuring distributed tracing for Docker environments (including Kubernetes)
+
+Because of the Kernel lockdown mode restrictions, your Docker and Kubernetes configuration files must mount 
+the `/sys/kernel/security/` volume for the **Beyla docker container** from your host system. This way Beyla can correctly 
+determine the Linux Kernel lockdown mode. Here's an example docker compose configuration, which ensures Beyla has sufficient
+information to determine the lockdown mode:
+
+```yaml
+version: '3.8'
+
+services:
+  ...
+  beyla:
+    image: grafana/beyla:latest
+    command:
+      - /beyla
+      - --config=/configs/beyla-config.yml
+    volumes:
+      - /sys/kernel/security:/sys/kernel/security
```

Comment:
If this volume mount is not provided, can Beyla continue working assuming that integrity mode is not set? Maybe show a warning telling that Beyla can't determine it and that the user show mount the volume to be 100% sure.

(The less mandatory configuration steps, the better)